projects/1.20/assets/integrated-scripting/integratedscripting/lang/zh_cn.json

```
@@ -0,0 +1,212 @@
+
+  "info_book.integratedscripting.writing.functions": "函数",
+  "info_book.integratedscripting.writing.functions.text1": "在动态联合的逻辑运算中，&2运算符&0是一等公民；不过就算如此，JavaScript也能使用函数创建它们。",
+  "info_book.integratedscripting.writing.functions.text2": "JavaScript中所有的函数创建方式都可用来创建&2运算符&0，&ofunction&r关键字和lambda函数均可。",
+  "info_book.integratedscripting.writing.functions.text3": "函数接受一定数量的参数。在动态联合中运行时，传入的JavaScript值均由动态联合的变量转化而来。",
+  "info_book.integratedscripting.writing.functions.text4": "JavaScript函数在&l变量卡&r中绑定为&2运算符&0，所需的操作过程与常量和变量无二：在脚本终端的编辑器中选中函数名，再绑定至&l变量卡&r。所有接受&2运算符&0的地方都能使用该值。",
+  "info_book.integratedscripting.writing.functions.text5": "比如说，创建接受单个&8物品&0参数、返回&9布尔型&0的函数，在读取箱子中物品列表时，就可用作过滤&2运算符&0。",
+  "info_book.integratedscripting.writing.functions.text6": "下方是不同类型函数的创建示例。",
+
+  "info_book.integratedscripting.writing.globals": "全局函数",
+  "info_book.integratedscripting.writing.globals.text1": "编写JavaScript代码时，可以通过全局变量&oidContext&r下的&oops&r字段访问所有动态联合运算符，这些运算符均视作函数。在写新运算符时，可以借此方式使用自带的运算符。",
+  "info_book.integratedscripting.writing.globals.text2": "比如，可以编写一个&8物品&0过滤器，以检验物品是否可堆叠，且堆叠数至少为16。示例如下。",
+  "info_book.integratedscripting.writing.globals.text3": "如需查询运算符的名称，可以在&l逻辑编程器&r中查看运算符说明里的全局名称，或在本书的“逻辑编程”条目下查找。",
+  "info_book.integratedscripting.writing.globals.text4": "通过&oidContext&r调用函数比较费空间，可以考虑将这些运算符存储到自定义的常量中去，示例见后。此外，还可以使用&o对象方法&r，这一内容详见下一条目。",
+
+  "info_book.integratedscripting.writing.methods": "对象方法",
+  "info_book.integratedscripting.writing.methods.text1": "使用全局函数可能会占用大部分空间，&l对象方法&r以紧凑性著称，可以很好地解决这一问题。",
+  "info_book.integratedscripting.writing.methods.text2": "在JavaScript中使用时，&8方块&0、&8物品s&0和&8流体s&0等对象类型均有对应&l方法&r。",
+  "info_book.integratedscripting.writing.methods.text3": "对象方法和普通的函数本质上区别不大，但方法的第一个参数与对象值联系极深。",
+  "info_book.integratedscripting.writing.methods.text4": "例如，全局函数&oitemstackStackable&r接受单个&8物品&0参数，并返回一个&9布尔型&0。此函数在&8物品&0对象中作为方法可用，名称为&ostackable&r；这一方法不接受参数。",
+  "info_book.integratedscripting.writing.methods.text5": "接受两个及以上参数的全局函数，均可用作对象的方法，其中各参数的次序均向前移动一位。比如，全局函数&oitemstackStrength&r接受一个&8物品&0和一个&8方块&0参数，它同样也可作为&8物品&0的方法使用，此时只接受单个&8方块&0参数。",
+
+  "info_book.integratedscripting.advanced": "进阶",
+
+  "info_book.integratedscripting.advanced.transient_storage": "瞬态存储",
+  "info_book.integratedscripting.advanced.transient_storage.text1": "动态联合的设计基于函数式编程的原则，因此基本不存在记录状态的手段，除非使用&l延迟器&r等外部方式绕过这一限制。",
+  "info_book.integratedscripting.advanced.transient_storage.text2": "JavaScript语言允许混用函数式编程和命令式编程，状态的记录也就在某种意义上成为了可能。",
+  "info_book.integratedscripting.advanced.transient_storage.text3": "具体说来，如果在函数外部声明了可变变量（比如说用了&olet&r），就可在调用的函数内修改它的值。",
+  "info_book.integratedscripting.advanced.transient_storage.text4": "这就是记录状态的方式了。统计函数调用次数、计算一系列特定值的平均数，这些任务都可用此方式完成。",
+  "info_book.integratedscripting.advanced.transient_storage.text5": "这些变量存在一个重大缺陷：它们的状态是瞬时的。也即，其值会随着重新插拔变量卡和重启服务器而重置。",
+  "info_book.integratedscripting.advanced.transient_storage.text6": "因此，这些值无法保证能永久存在。如果需要持久的数据存储方式，最好还是使用&l延迟器&r。",

```

transient暂态 --- persistent持久态
```suggestion
  "info_book.integratedscripting.advanced.transient_storage": "暂态存储",
  "info_book.integratedscripting.advanced.transient_storage.text1": "动态联合的设计基于函数式编程的原则，因此基本不存在记录状态的手段，除非使用&l延迟器&r等外部方式绕过这一限制。",
  "info_book.integratedscripting.advanced.transient_storage.text2": "JavaScript语言允许混用函数式编程和命令式编程，这一特性便可用于临时记录状态。",
  "info_book.integratedscripting.advanced.transient_storage.text3": "具体说来，如果在函数外部声明了可变变量（比如说用了&olet&r），该变量就可通过后续函数调用而修改。",
  "info_book.integratedscripting.advanced.transient_storage.text4": "基于这些暂态变量，可以做到统计函数调用次数、实时统计一系列值的平均数等工作。",
  "info_book.integratedscripting.advanced.transient_storage.text5": "需要强调的是，这些变量是暂态的而非持久的；具体而言，其值会随着重新插拔变量卡和重启服务器而重置。",
  "info_book.integratedscripting.advanced.transient_storage.text6": "因此，这些值无法保证能永久存在。如果需要持久的数据存储方式，最好还是使用&l延迟器&r。",
```